postgres_exporter/common/pg16/setup.sql

```diff
@@ -0,0 +1,463 @@
+
+v_throttle := make_interval(mins := p_throttle_minutes);
+
+SELECT COALESCE(max(gather_timestamp), '1970-01-01'::timestamptz) INTO v_gather_timestamp FROM monitor.pgbackrest_info;
+
+IF pg_catalog.pg_is_in_recovery() = 'f' THEN
+    IF ((CURRENT_TIMESTAMP - v_gather_timestamp) > v_throttle) THEN
+
+        -- Ensure table is empty 
+        DELETE FROM monitor.pgbackrest_info;
+
+        SELECT system_identifier into v_system_identifier FROM pg_control_system();
+
+        -- Copy data into the table directory from the pgBackRest into command
+        EXECUTE format( $cmd$ COPY monitor.pgbackrest_info (config_file, data) FROM program '/usr/bin/pgbackrest-info.sh %s' WITH (format text,DELIMITER '|') $cmd$, v_system_identifier::text );
+
+    END IF;
+END IF;
+
+RETURN QUERY SELECT * FROM monitor.pgbackrest_info;
+
+IF NOT FOUND THEN
+    RAISE EXCEPTION 'No backups being returned from pgbackrest info command';
+END IF;
+
+END 
+$function$;
+
+
+DROP FUNCTION IF EXISTS monitor.sequence_status();
+CREATE FUNCTION monitor.sequence_status() RETURNS TABLE (sequence_name text, last_value bigint, slots numeric, used numeric, percent int, cycle boolean, numleft numeric, table_usage text)  
+    LANGUAGE sql SECURITY DEFINER STABLE
+    SET search_path TO pg_catalog, pg_temp
+AS $function$
+
+/* 
+ * Provide detailed status information of sequences in the current database
+ */
+
+WITH default_value_sequences AS (
+    -- Get sequences defined as default values with related table
+    -- Note this subquery can be locked/hung by DDL that affects tables with sequences. 
+    --  Use monitor.sequence_exhaustion() to actually monitor for sequences running out
+    SELECT s.seqrelid, c.oid 
+    FROM pg_catalog.pg_attribute a
+    JOIN pg_catalog.pg_attrdef ad on (ad.adrelid,ad.adnum) = (a.attrelid,a.attnum)
+    JOIN pg_catalog.pg_class c on a.attrelid = c.oid
+    JOIN pg_catalog.pg_sequence s ON s.seqrelid = regexp_replace(pg_get_expr(ad.adbin,ad.adrelid), $re$^nextval\('(.+?)'::regclass\)$$re$, $re$\1$re$)::regclass
+    WHERE (pg_get_expr(ad.adbin,ad.adrelid)) ~ '^nextval\('
+), dep_sequences AS (
+    -- Get sequences set as dependencies with related tables (identities)    
+    SELECT s.seqrelid, c.oid
+    FROM pg_catalog.pg_sequence s 
+    JOIN pg_catalog.pg_depend d ON s.seqrelid = d.objid
+    JOIN pg_catalog.pg_class c ON d.refobjid = c.oid
+    UNION
+    SELECT seqrelid, oid FROM default_value_sequences
+), all_sequences AS (
+    -- Get any remaining sequences
+    SELECT s.seqrelid AS sequence_oid, ds.oid AS table_oid
+    FROM pg_catalog.pg_sequence s
+    LEFT JOIN dep_sequences ds ON s.seqrelid = ds.seqrelid
+)
+SELECT sequence_name
+    , last_value
+    , slots
+    , used
+    , ROUND(used/slots*100)::int AS percent
+    , cycle
+    , CASE WHEN slots < used THEN 0 ELSE slots - used END AS numleft
+    , table_usage
+FROM (
+     SELECT format('%I.%I',s.schemaname, s.sequencename)::text AS sequence_name
```

Comment:
Does this `format` need `pg_catalog` prefix?

Reply:
Will make note of this for a future update. Thanks!